sshd/terminal/terminal.go

```diff
@@ -462,6 +464,10 @@ func visualLength(runes []rune) int {
 			inEscapeSeq = true
 		default:
 			length++
+			kind := width.LookupRune(r).Kind()
+			if kind == width.EastAsianFullwidth || kind == width.EastAsianWide {
```

Comment:
What happens if we don't check the `kind`?

Reply:
If we don't check the `kind`, the cursor will move half steps than needed in fullwidth words. then leads to #336. 

According to [Halfwidth_and_fullwidth_forms](https://en.wikipedia.org/wiki/Halfwidth_and_fullwidth_forms) in wiki,
> Unicode assigns every code point an "East Asian width" property. 

> Terminal emulators can use this property to decide if a character should consume one or two "columns" when figuring out tabs and cursor position.

It's a property for every runes, there are ASCII characters in fullwidth (e.g. "ＡＢ" not "AB").
[Kind](https://github.com/golang/text/blob/v0.3.2/width/width.go#L31) indicates the type of width property.